### ISSUE TITLE
Incidents: add the ability to document notes on a reported incident

### ISSUE THREAD

**Issue by blestab** [HospitalRun/hospitalrun-frontend#2497]
## 🚀 Feature Proposal

Add the ability to document notes onto a reported incident.

## Motivation

During the incident's lifespan, there will be need to add multiple notes relating to the incident.

#### Incident Model 

Add a notes field which hold multiple notes objects

```
notes?: Note[]
```

#### Note Model 

Add a givenBy field which will store the currently logged in user writing the note. This will be an optional field.

```
givenBy?: string
```

#### Incident View/Edit 
On the incident view/edit form, at the bottom
- Add a Notes sections
   In the notes section:
   - Add a "New Note" button which opens a popup to add a new note to the incident
   - Display a table of already existing notes, showing the following fields: date, givenBy, note

![image](https://user-images.githubusercontent.com/5468546/101275649-d78bcd00-37af-11eb-8c37-1f5409cb43de.png)

**Comment by aguptamusic** [comment]
Hi @blestab, could I take this?

**Comment by blestab** [comment]
Yes you may @aguptamusic , thanks

**Comment by aguptamusic** [comment]
@blestab @jackcmeyer 
The design provided above for the "+ New Note" button doesn't match the style of the "+ Add New Note" button in the patient notes view. The "+ New Note" design above has a green background and looks similar to the "Resolve Issue" button, whereas the "+ Add New Note" has a white background that turns blue when hovering. Which should we use?

**Comment by blestab** [comment]
Thanks @aguptamusic please go ahead and use button label, functionality (display message if there are no notes) and design/styling that matches the Patient Notes view. In addition though, we need a table of notes for this one instead of how Patient Notes displays the list of notes, with the ability to delete/edit a note.

**Comment by aguptamusic** [comment]
@blestab Given that the core team seems to still be working out logging in functionality, it is not obvious what role the current user plays (Doctor, etc.). How should we work with the givenBy field in a Note? A few options we're thinking about:

1) Try to replicate App.tsx and use the current session and user object from a try/catch and handle the inevitable error case from RemoteDB.
2) Comment out the givenBy field and ignore it for now. 
3) Insert a dummy string in the field, making them all uniform for now.

What do you recommend we do?

**Comment by jackcmeyer** [comment]
You can get the current user from the redux store, which currently has mock data inside of it. You can see an example of how to do that in the report incident functionality.

**Comment by DrewGregory** [comment]
If a user updates a note with the "Edit" functionality, should the date/givenBy be updated?

**Comment by blestab** [comment]
I don't believe so as those are the create date and user. In future we could possibly look at introducing lastEditDate, and lastEditUser fields.

**Comment by DrewGregory** [comment]
Makes sense.

The patient module also has a notes section. Would you like us to also make the notes section for the patient module use the same Notes Table UI? Currently the patient notes tab is a list without support for edit/delete.

**Comment by blestab** [comment]
> Makes sense.
> 
> The patient module also has a notes section. Would you like us to also make the notes section for the patient module use the same Notes Table UI? Currently the patient notes tab is a list without support for edit/delete.

Good idea. That would be great for UX consistency; We will log a separate issue for it.
